scrapyjs/middleware.py

```diff
@@ -43,24 +52,26 @@ def from_crawler(cls, crawler):
         return cls(crawler, splash_base_url, slot_policy)
 
     def process_request(self, request, spider):
-        splash_options = request.meta.get('splash')
+        splash_options = self.get_splash_options(request, spider)
         if not splash_options:
             return
 
+        elif request.meta.get("_splash_processed"):
+            return
```

Comment:
this is because if you return new request from middleware it will go through all middleware chain again, so we need to disable infinite loop from occurring